Description: use GL_CLAMP_TO_EDGE for video texture mapping because most gfx
driver doesn't support GL_REPEAT when texture size is None-Power-Of-Tow(NPOT)
Author: Halley Zhao <halley.zhao at intel.com>
--- a/src/multimedia/qpaintervideosurface.cpp
+++ b/src/multimedia/qpaintervideosurface.cpp
@@ -361,8 +361,8 @@ QAbstractVideoSurface::Error QVideoSurfa
                     m_frame.bits() + m_textureOffsets[i]);
             glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
             glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-            glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
-            glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+            glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+            glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
         }
         m_frame.unmap();
     } else if (m_handleType != QAbstractVideoBuffer::QPixmapHandle && m_frame.isValid()) {
